tensorboard/webapp/metrics/views/main_view/common_selectors.ts

```diff
@@ -287,10 +289,49 @@ export const getPotentialHparamColumns = createSelector(
       sortable: true,
       movable: true,
       filterable: true,
+      hidable: true,
     }));
   }
 );
 
+export const getSelectableColumns = createSelector(
+  getPotentialHparamColumns,
+  getDashboardDisplayedHparamColumns,
+  (potentialColumns, currentColumns) => {
+    const currentColumnNames = new Set(currentColumns.map(({name}) => name));
+    return potentialColumns.filter((columnHeader) => {
+      return !currentColumnNames.has(columnHeader.name);
+    });
+  }
+);
+
+/** Returns a list of columns that have been sorted into logical groups.
+ *
+ * Column order: | RUN | experimentAlias | HPARAMs | other |
+ */
+export const getGroupedColumns = (
+  headersSelector: Selector<object, ColumnHeader[]>
```

Comment:
This is my opinion, feel free to push back.

We don't use the pattern of passing a selector to a selector anywhere else in the TB codebase. Due to this I expect this to confuse future devs coming to the project.
I would rather see this selectors projector moved to a utility function and shared by a series of selectors (not selector factories) created to select each type.

The result might look something like this
```
function groupHeaders(tableHeaders: ColumnHeader[], hparamHeaders: []) {
   // The projector function from the getGroupedColumns selector
}

export const selectGroupedRunTableHeaders = createSelector(
  getRunsTableHeaders,
  getDashboardDisplayedHparamColumns,
  groupHeaders
);

export const selectGroupedDataTableHeaders = createSelector(
  getDataTableHeaders,
  getDashboardDisplayedHparamColumns,
  groupHeaders
);
```

Reply:
> I would rather see this selectors projector moved to a utility function and shared by a series of selectors

This is indeed a much simpler abstraction than passing around selectors as function arguments. I created a `groupColumns` helper similar to what you described. I put it in `/widgets/data_table/utils.ts` because it doesn't really belong in the `/metrics` directory.

Reply:
I also moved the runs and metrics selectors to their respective directories, and added a simple selector test for each.